### PR TITLE
Add Hexagon IDE conforming packet syntax

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

In the programmers reference manual and the IDE, instruction packets are enclosed in curly brackets. E.g.
```c
{
<first instr>
<second instr>
}
```

But until now the plugin marked packets with other characters:

```c
// No UTF-8

?   jumpr R31
/   immext(0x1c00)
|   R11 = loc.start_pc
|   immext(0x1c00)
\   R12 = loc.start_sp
/   immext(0x1c00)
|   R13 = loc.start_param
\   R15 = 0x1
/   memw(R11+R2<<0x2) = R0
\   memw(R12+R2<<0x2) = R1    < endloop0

// UTF-8
?   jumpr R31
┌   immext(0x1c00)
│   R11 = loc.start_pc
│   immext(0x1c00)
└   R12 = loc.start_sp
┌   immext(0x1c00)
│   R13 = loc.start_param
└   R15 = 0x1
┌   memw(R11+R2<<0x2) = R0
└   memw(R12+R2<<0x2) = R1     ∎ endloop0
```

This PR enables the use of Hexagon IDE conforming packet markings.
This comes in very handy if you want to copy and parse some assembly into the IDE.

The result, after enabling the syntax with `plugins.hexagon.sdk`, is:
```c
?   jumpr R31
{   immext(0x1c00)
    R11 = loc.start_pc
    immext(0x1c00)
    R12 = loc.start_sp }
{   immext(0x1c00)
    R13 = loc.start_param
    R15 = 0x1 }
{   memw(R11+R2<<0x2) = R0
    memw(R12+R2<<0x2) = R1 }:endloop0
```

**Test plan**

Tests were added.

**Closing issues**

None

### CODE REVIEW

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by Rot127** [comment]
Conflicts come from the different timestamps in the header. Please let me know when you want to merge. I'll push updated timestamps then.

**Comment by wargio** [comment]
i guess we can merge, just resolve them

**Comment by Rot127** [comment]
After breaking everything during rebase, it should be ready once its green.

**Comment by Rot127** [comment]
```
[XX] db/formats/pdb idpdj
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '%RZ_CURL=1
idpdj~{}>$pdb
$pdb~file,guid,download
' bins/pdb/user32.dll
-- stdout
--- expected
+++ actual
@@ -1,3 +1,5 @@
+
+All done, errors in processing 1 file(s)
     "file": "user32.pdb",
     "guid": "15F778B3671D4EFDBDCDE79905308B792",
     "download": true

-- stderr
Attempting to download compressed pdb in /Users/runner/.local/share/rizin/pdb/user32.pdb/15F778B3671D4EFDBDCDE79905308B792/user32.pd_
Attempting to decompress pdb
/Users/runner/.local/share/rizin/pdb/user32.pdb/15F778B3671D4EFDBDCDE79905308B792: Is a directory
cab extractor exited with error 256
Falling back to uncompressed pdb
Attempting to download uncompressed pdb in /Users/runner/.local/share/rizin/pdb/user32.pdb/15F778B3671D4EFDBDCDE79905308B792/user32.pdb
```

Is the failing test my fault?
Couldn't find something in the issues, but it seems unrelated to my changes.

**Comment by Rot127** [comment]
Test succeeds on my Linux. Seems to be a problem with MacOS.

**Comment by XVilka** [comment]
```
[XX] db/formats/pyc pyc disasm
RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'pd 10
' bins/pyc/py37.pyc
-- stdout
--- expected
+++ actual
@@ -3,7 +3,8 @@
             ;-- <module>:
             0x0000002a      LOAD_CONST Multiline strings can be written
     using three "s, and are often used
-    as documentation. ; [00] ---- section size 3218 named module
+    as documentation.
+ ; [00] ---- section size 3218 named module
             0x0000002c      STORE_NAME             __doc__
             0x0000002e      LOAD_CONST                True
         ,=< 0x00000030      JUMP_IF_FALSE_OR_POP                  10

-- stderr
Undefined type in free_object (0)
WARNING: No calling convention defined for this file, analysis may be inaccurate.
```

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@ret2libc take a look please and merge if there are no objections.

File: librz/include/rz_util/rz_str.h [hunk 1]
```diff
@@ -149,13 +149,16 @@ static inline const char *rz_str_word_get_next0(const char *str) {
...
+RZ_API RZ_BORROW void rz_str_trim(RZ_INOUT char *str);
+RZ_API RZ_BORROW void rz_str_trim_char(RZ_INOUT char *str, const char c);
 RZ_API char *rz_str_trim_dup(const char *str);
 RZ_API char *rz_str_trim_lines(char *str);
 RZ_API void rz_str_trim_head(char *str);
+RZ_API RZ_BORROW void rz_str_trim_head_char(RZ_INOUT char *str, const char c);
 RZ_API const char *rz_str_trim_head_ro(const char *str);
 RZ_API const char *rz_str_trim_head_wp(const char *str);
 RZ_API RZ_BORROW char *rz_str_trim_tail(RZ_NONNULL char *str);
+RZ_API RZ_BORROW void rz_str_trim_tail_char(RZ_INOUT char *str, const char c);
```
Comment (wargio) [inline librz/include/rz_util/rz_str.h]:
```suggestion
RZ_API void rz_str_trim(RZ_NONNULL RZ_INOUT char *str);
RZ_API void rz_str_trim_char(RZ_NONNULL RZ_INOUT char *str, const char c);
RZ_API char *rz_str_trim_dup(const char *str);
RZ_API char *rz_str_trim_lines(char *str);
RZ_API void rz_str_trim_head(RZ_NONNULL char *str);
RZ_API void rz_str_trim_head_char(RZ_NONNULL RZ_INOUT char *str, const char c);
RZ_API const char *rz_str_trim_head_ro(const char *str);
RZ_API const char *rz_str_trim_head_wp(const char *str);
RZ_API RZ_BORROW char *rz_str_trim_tail(RZ_NONNULL char *str);
RZ_API void rz_str_trim_tail_char(RZ_NONNULL RZ_INOUT char *str, const char c);
```

File: librz/util/str_trim.c [hunk 2]
```diff
@@ -126,9 +135,59 @@ RZ_API RZ_BORROW char *rz_str_trim_tail(RZ_NONNULL char *str) {
...
-// Removes spaces from the head of the string, and zeros out whitespaces from
-// the tail of the string. The string is changed in place.
-RZ_API void rz_str_trim(char *str) {
+/**
+ * \brief Removes the the character \p c from the beginning of a string.
+ *
+ * \param str The string.
+ * \param c The character to be removed.
+ */
+RZ_API RZ_BORROW void rz_str_trim_head_char(RZ_INOUT char *str, const char c) {
```
Comment (wargio) [inline librz/util/str_trim.c]:
```suggestion
RZ_API void rz_str_trim_head_char(RZ_NONNULL RZ_INOUT char *str, const char c) {
```

File: librz/util/str_trim.c [hunk 3]
```diff
@@ -126,9 +135,59 @@ RZ_API RZ_BORROW char *rz_str_trim_tail(RZ_NONNULL char *str) {
...
+	}
+}
+
+/**
+ * \brief Removes the the character \p c from the end of a string.
+ *
+ * \param str The string.
+ * \param c The character to be removed.
+ */
+RZ_API RZ_BORROW void rz_str_trim_tail_char(RZ_INOUT char *str, const char c) {
```
Comment (wargio) [inline librz/util/str_trim.c]:
```suggestion
RZ_API void rz_str_trim_tail_char(RZ_NONNULL RZ_INOUT char *str, const char c) {
```

File: librz/util/str_trim.c [hunk 4]
```diff
@@ -126,9 +135,59 @@ RZ_API RZ_BORROW char *rz_str_trim_tail(RZ_NONNULL char *str) {
...
+	}
+}
+
+/**
+ * \brief Removes the character \p c from the beginning and end of a string.
+ *
+ * \param str The string to trim.
+ * \param c The character to remove.
+ */
+RZ_API RZ_BORROW void rz_str_trim_char(RZ_INOUT char *str, const char c) {
```
Comment (wargio) [inline librz/util/str_trim.c]:
```suggestion
RZ_API void rz_str_trim_char(RZ_NONNULL RZ_INOUT char *str, const char c) {
```

File: librz/util/str_trim.c [hunk 5]
```diff
@@ -126,9 +135,59 @@ RZ_API RZ_BORROW char *rz_str_trim_tail(RZ_NONNULL char *str) {
...
+	rz_str_trim_tail_char(str, c);
+}
+
+/**
+ * \brief Removes whitespace characters (space, tab, newline etc.)
+ * from the beginning and end of a string.
+ *
+ * \param str The string to trim.
+ */
+RZ_API RZ_BORROW void rz_str_trim(RZ_INOUT char *str) {
```
Comment (wargio) [inline librz/util/str_trim.c]:
```suggestion
RZ_API void rz_str_trim(RZ_NONNULL RZ_INOUT char *str) {
```

**Review by wargio — CHANGES_REQUESTED** [review]

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

File: librz/include/rz_asm.h [hunk 6]
```diff
@@ -118,6 +118,7 @@ typedef struct rz_asm_t {
 	bool immdisp; // Display immediates with # symbol (for arm/hexagon architectures). false = show hashs
 	bool immsign; // Print signed immediates as negative values, not their unsigned representation.
 	bool utf8; // Flag for plugins: Use utf-8 characters.
+	bool hex_sdk; // Hexagon arch only. Print packet syntax in hexagon-objdump style.
```
Comment (XVilka) [inline librz/include/rz_asm.h:121]: In the future, we should group implementation-specific options together.

Comment (wargio) [reply librz/include/rz_asm.h:121]: i think we should have custom evars + context for this stuff.

**Pull request merged**